docs/logging.md

```diff
@@ -0,0 +1,167 @@
+---
+id: logging
+title: Logging
+---
+
+### Logging
+
+Almost every application nowadays does logging, whether it's logging of business events, history of some state of 
+application or even debug record sequences. We log to files, distributed systems, standard outputs of OSes, and we 
+got used to it so much it seems really simple to us. Many existing libraries provide many ways of logging from 
+your application's code, either automatic or manual.
+
+### What's wrong?
+
+First, logging is, in fact, a side effect. Whether we log anything to a standard output, or a file - it's an interaction 
+with the outside world ('outside' relative to our program). Thus, logging as a process can be represented as an effect and operations - as an algebra.  
+Second, logging itself seems easy, but stuff around it - not so much. MDC, contexts, structured logging - all these 
+things are very helpful on a day-to-day basis, allowing us to read and write our logs, making them clean, comprehensible 
+and useful for many purposes. But there are few simple ways to provide all this functionality without bleeding and 
+screaming from seeing over-engineered solutions.
+
+### The solution: Logging
+
+Tofu offers an easy and understandable abstraction over logging (it's even named like this!) but keeping in mind that 
+logging is an effect that can be composed.  
+It looks very similar to interfaces we all are used to (from SLF4J and other logging libraries and facades), introducing some new concepts:
```

Comment:
The conclusion of the paragraph is almost identical, the only difference is clarifying what actually is the new feature. Maybe collapse it with this sentence? That would additionally provide the reader with an information of what new concepts to look for, compared to "old" interfaces